### ISSUE TITLE
[Bug]: Server download URL returning 404

### ISSUE THREAD

**Issue by tvories** [advplyr/audiobookshelf#714]
### Describe the issue

When trying to click the "download" button on a track on the app, a 404 is returned.  This is causing crashes in the android app when trying to download local to the phone.  Oddly, streaming works fine.

![image](https://user-images.githubusercontent.com/13934524/173171968-77ed5b35-4c2b-4587-8503-da09ac3372d0.png)

result:
`Cannot GET /s/item/li_6vmhwv7q2tru6rxiei/Pachinko-Part01.mp3`

### Steps to reproduce the issue

1. Use docker hosted audiobookshelf leveraging NFS storage
2. Scan library files
3. Open a book and try to download a single file from the book
4. Receive a 404

### Audiobookshelf version

v2.0.21

### How are you running audiobookshelf?

Other - Kubernetes

**Comment by jmt-gh** [comment]
I'm not able to reproduce this. 

Can you try this?

1. Go to the settings gear -> user -> select your user -> API key.
2. Open a new tab
3. browse to: http://<YOURDOMAIN>.com/s/item/li_6vmhwv7q2tru6rxiei/Pachinko-Part01.mp3?token=<YOUR API TOKEN>

If successful, it should start streaming the part directly in your browser (at least that's what I see on firefox). As well, I'm curious -- when you hover over the download icon -- what URL is it directing you to? I'm curious if this is a DNS-esque issue

**Comment by helio4k** [comment]
I am just wondering if you are going through a reveres proxy? I hade a similar experience long time ago but it was due to misconfiguration of my Nginx Proxy manager.

**Comment by tvories** [comment]
> I am just wondering if you are going through a reveres proxy? I hade a similar experience long time ago but it was due to misconfiguration of my Nginx Proxy manager.

I do run my setup through a reverse proxy, but I tried this with both the reverse proxy and direct. The error I posted above is from the direct connection (no reverse proxy).

**Comment by tvories** [comment]
> I'm not able to reproduce this.
> 
> Can you try this?
> 
>     1. Go to the settings gear -> user -> select your user -> API key.
> 
>     2. Open a new tab
> 
>     3. browse to: http://.com/s/item/li_6vmhwv7q2tru6rxiei/Pachinko-Part01.mp3?token=
> 
> 
> If successful, it should start streaming the part directly in your browser (at least that's what I see on firefox). As well, I'm curious -- when you hover over the download icon -- what URL is it directing you to? I'm curious if this is a DNS-esque issue

I copied that API key manually and replaced the token field with that API token copied from the user account.  Same issue.  When I right-click the link to the direct file and copy it, it is the same link with the same token.

Direct link URL (no reverse proxy): http://localhost:9911/s/item/li_6vmhwv7q2tru6rxiei/Pachinko-Part01.mp3?token=my-user-token

I also tried creating a new user and tried the same link logged in as the other user.  Same result

**Comment by tvories** [comment]
I went back versions until this worked again and the URLs work (with reverse proxy) as of 2.0.14.  Something in 2.0.15 breaks these download URLs

**Comment by tvories** [comment]
I have a feeling it's related to this: https://github.com/advplyr/audiobookshelf/issues/529

**Comment by advplyr** [comment]
Are you able to re-create this issue on the latest release 2.0.21?

**Comment by advplyr** [comment]
I see now that you originally had the issue with the latest version.
The last time the static router was updated was when adding support for downloading single file audiobooks.
Single file audiobooks are handled differently because they are put into the root of the library folder so they don't have a folder. https://github.com/advplyr/audiobookshelf/commit/6cfe583535217a8f4aa31329e57fa01208e282b4

I noticed in your screenshot that you don't have a "Manage Tracks" button.

![image](https://user-images.githubusercontent.com/67830747/173256310-ae9910ed-71d0-4c2f-bf0d-4d7037e13b23.png)

That button is hidden when you have the "isFile" flag set on the audiobook. For some reason that flag was set on your audiobook.
Did you originally have the audio file inside the library folder root then moved it into a subfolder?

**Comment by tvories** [comment]
I have gone through a few different iterations of my library.  Originally, my books were mounted to: `/audiobooks`.  Then I moved to `/media/books`.  I believe when I was testing fixes, I thought maybe it was related to the directory level, so when I was doing this troubleshooting, I went back to `/audiobooks`.   2.0.21 did not work with /audiobooks or /media/books.  I am back on 2.0.14 and downloads are working great.

**Comment by tvories** [comment]
In the process of troubleshooting, I have destroyed and re-created my library many times.  Every item in my library is unable to be downloaded because of the 404.

I'll try removing the item and re-scanning after going to 2.0.21

**Comment by advplyr** [comment]
If you don't see the "Manage Tracks" button on the tracks dropdown of the library item then you won't be able to download.

**Comment by tvories** [comment]
I updated to 2.0.21 and removed the library item.  I re-scanned, went to the library item, and same results - the book is not recognized as a multi-file audiobook.

Is there something in the logs I can look for when it does a scan to see why it's not getting the correct type of book?

**Comment by tvories** [comment]
This definitely has something to do with the way my NFS volume is mounted.  I mounted a direct file to my deployment, copied the same Pachinko book, and it comes up with the "Manage Tracks" option.  I will mess around with NFS permissions a bit and see if I can determine which ones are throwing things off

**Comment by advplyr** [comment]
I'm not sure this is related to NFS, this could be a bug with how the scanner detects single file audiobooks in the library folder root.
What is the folder path and the library item path? I can re-create your directory structure for that audiobook that is scanned in wrong to test it.

**Comment by Thovi98** [comment]
I’ve same issue, although I didn’t have it a few released before (I discovered it in 2.0.20). 
I can play any audiobook, but cannot download them. 
I don’t know how I can help resolving this issue ?

**Comment by tvories** [comment]
@Thovi98 are you using NFS for storage?

**Comment by Thovi98** [comment]
I don’t think so, at least I don’t have nfs- related packages installed and I don’t know understand exactly what it does, so I don’t think (I’m still a newbie) 😁

**Comment by simonbh** [comment]
I experienced this issue for one of my audiobooks.  Single file download and playback returned a HTTP 404 message.
Running version 2.0.23 via docker-compose.  My audiobooks are stored on a TrueNAS host via NFS.  

I corrected the issue by changing the "&" character to "and" in the audiobook file name and author directory.
For example:
From:
/audiobooks/James Smith & Kevin Smith/Great Audiobook by James Smith & Kevin Smith.mp3
To:
/audiobooks/James Smith and Kevin Smith/Great Audiobook by James Smith and Kevin Smith.mp3

I then re-scanned my library, The new directory/file was then added.  I then manually removed the old entry.  After this, download and playback worked as expected.
